File: mega/src/main/java/com/example/mega/billing/model/Bill.java
package com.example.mega.billing.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;

import lombok.Data;

@Data
@Entity
@Table(name = "bills")
public class Bill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bill_id")
    private Integer billId;

    @Column(name = "booking_id", unique = true)
    private Integer bookingId;

    @NotNull(message = "Total amount cannot be null")
    @Column(name = "total_amount")
    private BigDecimal totalAmount;

    @Column(name = "tax_amount")
    private BigDecimal taxAmount;

    @Column(name = "discount_amount")
    private BigDecimal discountAmount;

    @Column(name = "payment_status")
    private String paymentStatus;

    @Column(name = "payment_method")
    private String paymentMethod;

    @Column(name = "bill_date")
    private LocalDateTime billDate;
}

File: frontend/src/types/bill.ts
export interface Bill {
    billId: number;
    bookingId: number;
    totalAmount: number;
    taxAmount: number;
    discountAmount: number;
    paymentStatus: string;
    paymentMethod: string;
    billDate: string;
}

export interface BillDTO {
    billId?: number;
    bookingId?: number;
    totalAmount?: number;
    taxAmount?: number;
    discountAmount?: number;
    paymentStatus?: string;
    paymentMethod?: string;
    billDate?: string;
}

export interface CreateBillResponse extends Bill {}
export interface GetBillByIdResponse extends Bill {}
export interface GetBillByBookingIdResponse extends Bill {}
export interface UpdateBillResponse extends Bill {}
export interface GetAllBillsResponse extends Array<Bill> {}
